terraform_files/vsphere/variables-vsphere.tf

```diff
@@ -0,0 +1,91 @@
+//////
+// vSphere variables
+//////
+
+variable "vsphere_vcenter" {
+  type        = string
+  description = "vSphere vcenter server ip address or fqdn (vCenter server name for vSphere API operations)"
+}
+
+variable "vsphere_username" {
+  type        = string
+  description = "vSphere vcenter server username"
+}
+
+variable "vsphere_password" {
+  type        = string
+  description = "vSphere vcenter server username"
+}
+
+variable "cluster_name" {
+  type = string
+  description = <<EOF
+AI cluster name
+All the resource will be located under a folder with this name
+and tagged with the cluster name
+The resources should be associate with this name for easy recognition.
+EOF
+}
+
+variable "masters_count" {
+  type = string
+  default = "3"
+  description = "The number of master nodes to be created."
+}
+
+variable "iso_download_path" {
+  type        = string
+  description = "The ISO path (We have to upload this file to the vsphere)"
+  default = "dummy"
+}
+
+variable "vsphere_cluster" {
+  type        = string
+  description = "vSphere cluster name"
+}
+
+variable "vsphere_datacenter" {
+  type        = string
+  description = "vSphere data center name"
+}
+
+variable "vsphere_datastore" {
+  type        = string
+  description = "vSphere data store name"
+}
+
+variable "vsphere_network" {
+  type        = string
+  description = "vSphere publicly accessible network for cluster ingress and access. e.g VM Network"
+}
+
+///////////
+// Control Plane machine variables
+///////////
+variable "master_memory" {
+  type = number
+  default = 16984
+  description = "The size of the master's virtual machine's memory, in MB"
+}
+
+variable "master_disk_size_gib" {
+  type = number
+  default = 120
+  description = "The size of the master's disk, in GB"
+}
+
+variable "master_vcpu" {
```

Comment:
Some of these variables are duplicated.
Do we really need to redefine these common variables?
https://github.com/openshift/assisted-test-infra/blob/51d5d1c28822dba48ac56dd4210117d0b606a272/terraform_files/baremetal/variables-libvirt.tf#L131-L135
Could it be a shared variables file?

Reply:
In the terraform controller we translate between the config and the terraform variables. 
This controller just pass the config as a terraform variables like this:
config = {**self._config.get_all(), **self._cluster_config.get_all(), "cluster_name": self.cluster_name}
and then filter the irrelevant variables. IMO more elegant.

Reply:
Need to refactor the terraform controller as well but not as part of this epic

Reply:
Please open a ticket

Reply:
https://issues.redhat.com/browse/MGMT-7232?filter=-2